extension/bootstrap.js

```diff
@@ -232,7 +275,33 @@ const windowListener = {
 
 this.install = function(data, reason) {};
 
-this.startup = function(data, reason) {
+this.startup = async function(data, reason) {
+  studyUtils.setup({
+    studyName: config.study.studyName,
+    endings: config.study.endings,
+    addon: { id: data.id, version: data.version },
+    telemetry: config.study.telemetry,
+  });
+  studyUtils.setLoggingLevel(config.log.studyUtils.level);
+  const variation = await chooseVariation();
+  studyUtils.setVariation(variation);
+
+  // TODO Import config.modules?
+
+  if ((REASONS[reason]) === "ADDON_INSTALL") {
```

Comment:
Agreed, this is needlessly complicated and can be simplified to use only one of the methods.
It seems to be that using the `REASON[reason] === "ADDON_INSTALL` way is cleaner, since there is no need to use a "reason string" as a key in the map.